core/common/src/main/java/alluxio/AbstractClient.java

```diff
@@ -180,7 +180,7 @@ public synchronized void connect() throws AlluxioStatusException {
       try {
         mAddress = getAddress();
       } catch (UnavailableException e) {
-        if (!retryPolicy.attemptRetry()) {
+        if (!retryPolicy.attempt()) {
```

Comment:
I think this can be simplified with `while (retryPolicy.attempt())` and just doing a `continue` here

Reply:
Done